### PR TITLE
Improved Contributing page and minor grammatical fixes to index.md

### PR DESCRIPTION
**Pull request by johnzl-777**
Modified `contrib.md` with the following changes:
* slightly rewording on first paragraph
* The CLI tool is now mentioned much earlier, before all examples using it
* Wording to encourage contributions to documentation AND unit tests
* Another section specifically for unit testing using the CLI tool

Modified `index.md` with the following changes:
* minor grammatical improvements
* Another bullet point at the bottom to encourage contributors to follow the contribution guide and make a pull request with their code

### CODE REVIEW

File: docs/src/contrib.md [hunk 1]
```diff
@@ -2,23 +2,31 @@
 
 If you are interested in contributing to this package,
 please consider going through this guide to help make your
-developing workflow smooth.
+development workflow as smooth as possible.
 
-Contributing to the documentation is always a good start to get yourself familiar
+Contributing to documentation and unit tests are always great ways to get yourself familiar
 with the community and workflows.
```
Comment (fanglifl) [inline docs/src/contrib.md:8]: "unit tests" is a little unfamiliar to me. @Roger-luo , could you take a look of this point?

File: docs/src/contrib.md [hunk 2]
```diff
@@ -2,23 +2,31 @@
...
+terminal to see the help message. or run `.ci/run <command> -h`
+to see the help message of each command.
+Below are some common use cases.
+
 ## Documentation
 
 ## Setup Documentation
 
-If you are editing the documentation, you can use the `serve` command:
+If you're editing the documentation, you can use the `serve` command:
```
Comment (fanglifl) [inline docs/src/contrib.md:23]: in scientific writing, we usually do not write abbreviation, so I think "you are" might be better?

File: docs/src/contrib.md [hunk 3]
```diff
@@ -32,7 +40,7 @@ which will run the build:
 Due to the "ancient" technology used by Documenter,
 it cannot render single page while editing, which
 causes the `doc serve` command to be very slow
-when you are editing. We provide a light-weight build setup
+when you're editing. We provide a light-weight build setup
```
Comment (fanglifl) [inline docs/src/contrib.md:43]: see comment as above

File: docs/src/index.md [hunk 4]
```diff
@@ -134,4 +134,5 @@ end
 
 ## Have Suggestions or Interested in Contributing?
 
-- [file an issue](https://github.com/QuEraComputing/Bloqade.jl/issues) to report a bug or request a feature.
+- [File an issue](https://github.com/QuEraComputing/Bloqade.jl/issues) to report a bug or request a feature.
+- Check out the [Contributing to Bloqade](@ref) guide if you'd like to make your own changes and make a [Pull Request](https://github.com/QuEraComputing/Bloqade.jl/pulls) when you'd like your work to be reviewed
```
Comment (fanglifl) [inline docs/src/index.md:138]: abbreviation issue?

**Review by fanglifl — CHANGES_REQUESTED** [review]
good changes. I only have a few minor comments

File: docs/src/contrib.md (hunk 2 above)
Comment (johnzl-777) [reply docs/src/contrib.md:23]: Got it! Apologies on my end, I'll go ahead and undo the abbreviations

**Review by fanglifl — APPROVED** [review]

**Pull request merged**